drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -110,45 +273,80 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     at86rf2xx_set_addr_short(dev, &addr_long.uint16[ARRAY_SIZE(addr_long.uint16) - 1]);
 
     /* set default channel */
-    at86rf2xx_set_chan(dev, AT86RF2XX_DEFAULT_CHANNEL);
+    at86rf2xx_set_chan(dev, at86rf2xx_default_channels[dev->base.dev_type]);
     /* set default TX power */
     at86rf2xx_set_txpower(dev, AT86RF2XX_DEFAULT_TXPOWER);
     /* set default options */
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_AUTOACK, true);
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, true);
 
     static const netopt_enable_t enable = NETOPT_ENABLE;
-    netdev_ieee802154_set(&dev->netdev, NETOPT_ACK_REQ,
+    netdev_ieee802154_set(&dev->base.netdev, NETOPT_ACK_REQ,
                           &enable, sizeof(enable));
 
     /* enable safe mode (protect RX FIFO until reading data starts) */
     at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_2,
                         AT86RF2XX_TRX_CTRL_2_MASK__RX_SAFE_MODE);
-#ifdef MODULE_AT86RF212B
-    at86rf2xx_set_page(dev, AT86RF2XX_DEFAULT_PAGE);
+
+    switch (dev->base.dev_type) {
+#if IS_USED(MODULE_AT86RF212B)
+        case AT86RF2XX_DEV_TYPE_AT86RF212B: {
+            at86rf2xx_set_page(dev, AT86RF212B_DEFAULT_PAGE);
+            break;
+        }
 #endif
+    }
 
-#if !defined(MODULE_AT86RFA1) && !defined(MODULE_AT86RFR2)
-    /* don't populate masked interrupt flags to IRQ_STATUS register */
-    uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_CTRL_1);
-    tmp &= ~(AT86RF2XX_TRX_CTRL_1_MASK__IRQ_MASK_MODE);
-    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_1, tmp);
+    switch (dev->base.dev_type) {
+        default:;
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_CTRL_1);
+            tmp &= ~(AT86RF2XX_TRX_CTRL_1_MASK__IRQ_MASK_MODE);
+            at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_1, tmp);
```

Comment:
Again `uint8_t tmp` needs a scope to live in. You could use a forward declaration before the `switch()` or add this to its own scope by wrapping it into a pair of curly braces.

Reply:
Additionally, you code add pseudo modules `at86rf2xx_spi` and `at86rf2xx_periph`. The first would be always added if any of the SPI attached variants is used, and `at86rf2xx_periph` would be added if any of the peripheral variants (RFA1, RFR2) is used.

That would allow you to wrap the code above into:

```C
#if IS_USED(MODULE_AT86RF2XX_SPI)

#endif
```

As a result, the ROM wouldn't increase if only the periph variants are used.